explorer/testdata/operators/mod_builtin.carbon

```diff
@@ -7,13 +7,18 @@
 // RUN: %{explorer} --parser_debug --trace_file=- %s 2>&1 | \
 // RUN:   %{FileCheck} --match-full-lines --allow-unused-prefixes %s
 // AUTOUPDATE: %{explorer} %s
+// CHECK: Interface: 1
+// CHECK: Op: 1
 // CHECK: result: 0
 
 package ExplorerTest api;
 
-// TODO: T:! Mod
-fn DoMod[T:! ModWith(.Self) where .Result == .Self](x: T, y: T) -> T { return x % y; }
-
 fn Main() -> i32 {
-  return DoMod(4, 2);
+  var lhs: i32 = 3;
+  var rhs: i32 = 2;
```

Comment:
```suggestion
  var lhs: i32 = 8;
  var rhs: i32 = 3;
```
Just suggesting a testcase that's less likely to pass by chance (eg, because `%` always returns `1`, or is accidentally using `-` rather than `%`).

Reply:
I'll accept the changes, but I would note we should and do have tests covering more inputs.